index.bs

```diff
@@ -601,12 +601,12 @@ function getParentAsPromise(entry) {
 <xmp class=idl>
 interface FileSystemDirectoryEntry : FileSystemEntry {
     FileSystemDirectoryReader createReader();
-    void getFile(optional USVString? path,
-                 optional FileSystemFlags options,
+    void getFile(optional USVString? path = null,
```

Comment:
Ok, I looked at the WebIDL Spec... it seems you can do:

```suggestion
    void getFile(optional USVString path = "",
```

Reply:
Won't this also require `[TreatNullAs=EmptyString]` to avoid the string `"null"` being passed if called with `null`?

Reply:
No, because `null` will just be "null"  :)  That is, there won't be any magic. Would that be a problem?

Reply:
In Chrome, it had special handling for `null` as per spec (i.e. null, undefined, and "" were treated identically), hence the current spec.

Verified by (manual) tests:

https://github.com/web-platform-tests/wpt/blob/master/entries-api/support.js#L61

https://github.com/web-platform-tests/wpt/blob/master/entries-api/filesystemdirectoryentry-getFile-manual.html#L31

Reply:
I understand, but I guess I'm saying that Chrome probably shouldn't do the special handling because it translates to "magic" handling of `null` and `undefined`. I believe that is undesirable (which is why `[TreatNullAs=EmptyString]` is marked as an undesirable feature in WebIDL). 

Instead, it should just let the IDL layer deal with it, making it consistent with other Web APIs.

Maybe we could get a second opinion in case I'm totally wrong (which is likely)? @domenic?

Reply:
Remember, this spec exists to document reality, not define a desirable API surface.

Reply:
Sure, I get that too. But a single implementation is not a standard: If Chrome is doing funky things, we should fix Chrome rather than impose the behavior on other implementations (specially during incubation, whereby the API is just an idea and not any sort of "reality")?

Reply:
I mean, I agree with both of you. In general it's better to let Web IDL handle it. But this spec is basically just documentation for a many-year-old Chrome-only API that previously had no spec. Realistically there have been no signs of other vendors implementing it. I'm not sure why it's in WICG; maybe for some IPR protections. But I don't think we should be seriously considering changing the normative behavior here.

I especially don't think we should be considering such changes in a pull request that is meant to be a non-normative update to match Web IDL. In particular I think it would have been ideal to only do the dictionary changes in this PR, and then have a separate PR for the USVStrings to better match reality.

Reply:
> Realistically there have been no signs of other vendors implementing it.

Uh... it's implemented in Edge, Firefox and Safari. That's why I wrote the doc - the other browsers were copying Chrome's implementation without a spec.

Reply:
Sorry, I was thinking of the other filesystem API! In that case, yeah, we'd want to have a completely separate discussion, with all involved stakeholders, to see if they're willing to invest effort in changing away from what's implemented.

Reply:
You know what we need? [Another file system API... ](https://github.com/WICG/native-file-system) ! 😄 

So... in this current PR, the changes are:

1. Give the optional dictionary argument a default (`options = {}`)
2. Since it's a second optional argument, give the first optional argument a default as well (`path = null`)
3. Eliminate redundant algorithmic prose as a result of (2) 

I think that's the minimal change to _"only do the dictionary changes in this PR"_ ?

Reply:
I'm unclear why you need to give the first optional argument a default because you've done so for the second one, i.e. I don't understand (2).

Reply:
Hmmm. I thought (2) was required by Web IDL but I'm not finding the text. I likely misread something in https://heycam.github.io/webidl/#dfn-optional-argument - thanks for pointing out my error!